pybliometrics/scopus/serial_search.py

```diff
@@ -12,37 +12,46 @@ def results(self):
         """
         out = []
         search_results = self._json['serial-metadata-response'].get('entry', [])
-        for i in search_results:
+        for result in search_results:
             # OrderedDict to populate with individual serial data
             obs = OrderedDict()
-            for key in i:
-                if key != '@_fa' and i[key]:
-                    if key == 'subject-area':
-                        subject_data = _merge_subject_data(i[key])
-                        obs['subject_area_codes'] = subject_data[0]
-                        obs['subject_area_abbrevs'] = subject_data[1]
-                        obs['subject_area_names'] = subject_data[2]
-                    elif key == 'SNIPList' or key == 'SJRList':
-                        for j in _retrieve_source_rankings(i[key]):
-                            obs[j[0]] = j[1]
-                    elif key == 'citeScoreYearInfoList':
-                        for j in _retrieve_cite_scores(i[key]):
-                            obs[j[0]] = j[1]
-                    elif key == 'link':
-                        for j in _retrieve_links(i['link']):
-                            obs[j[0]] = j[1]
-                    elif key == 'yearly-data':
-                        if i['yearly-data'].get('info'):
-                            time_data = _retrieve_yearly_data(i['yearly-data']['info'])
-                            for j in time_data:
-                                obs[j[0]] = j[1]
-                    else:
-                        obs[key] = i[key]
+            for key, value in result.items():
+                if key == '@_fa':
+                    continue
+                if key == 'subject-area':
+                    if not value:
+                        continue
+                    subject_data = _merge_subject_data(value)
+                    obs['subject_area_codes'] = subject_data[0]
+                    obs['subject_area_abbrevs'] = subject_data[1]
+                    obs['subject_area_names'] = subject_data[2]
+                elif key == 'SNIPList' or key == 'SJRList':
+                    if not value:
+                        continue
+                    for j in _retrieve_source_rankings(value):
+                        obs[j[0]] = j[1]
+                elif key == 'citeScoreYearInfoList':
+                    if not value:
+                        continue
+                    for j in _retrieve_cite_scores(value):
+                        obs[j[0]] = j[1]
+                elif key == 'link':
+                    if not value:
+                        continue
+                    for j in _retrieve_links(value):
+                        obs[j[0]] = j[1]
+                elif key == 'yearly-data':
+                    if not value:
+                        continue
+                    time_data = _retrieve_yearly_data(value.get('info'))
+                    for j in time_data:
```

Comment:
I think this might break and result in an attribute error if the returned value is `None` (which is the default from `.get()`). Do we have reason to believe that the "info" key might be missing even though "yearly-data" is present? If yes, `.get()`'s default should be an empty list. If not, let's simply use `value['info']`.